[TGBOT-296] Create http client customer gateway implementation

Sav4ykVladzislau opened the pull request:
https://schneidervs.atlassian.net/browse/TGBOT-296

Sav4ykVladzislau committed:
Create bean Customer

Sav4ykVladzislau committed:
Create HTML customer-form

Sav4ykVladzislau committed:
Create CustomerGateway and CustomerController

Sav4ykVladzislau committed:
Update Customer.java

Sav4ykVladzislau committed:
Update customer-form.html

Sav4ykVladzislau committed:
change customer-form.html

Sav4ykVladzislau committed:
Merge remote-tracking branch 'origin/main'

Sav4ykVladzislau committed:
Merge remote-tracking branch 'origin/main'

Sav4ykVladzislau committed:
Merge branch 'main' into TGBOT-289-customer-methods

Sav4ykVladzislau committed:
Change directory

Sav4ykVladzislau committed:
Change customer-form.html

Sav4ykVladzislau committed:
Create mapper and spring context class

Sav4ykVladzislau committed:
Create test for userController

Sav4ykVladzislau committed:
Change pom file

Sav4ykVladzislau committed:
Change file

Sav4ykVladzislau committed:
Merge branch 'main' into TGBOT-289-customer-methods

# Conflicts:
#	web-ui/pom.xml

Sav4ykVladzislau committed:
Change dependency from pom.xml

Sav4ykVladzislau committed:
Moved to the correct directory
and create test for save and delete method

Sav4ykVladzislau committed:
add method deleteCustomerById

Sav4ykVladzislau committed:
change customer-form.html

Sav4ykVladzislau committed:
Update customer-form.html

add empty line

Sav4ykVladzislau committed:
Change file name by SpringContext

Sav4ykVladzislau committed:
Change returned object in CustomerGateway Customer -> Optional

Sav4ykVladzislau committed:
Change returned object in CustomerGateway Customer -> Optional
and change test

Sav4ykVladzislau committed:
Change returned object in CustomerGateway Customer -> Optional
and change test

Sav4ykVladzislau committed:
Change customers-form.html to customer-form & remove HttpServletRequest r row

Sav4ykVladzislau committed:
Change Optional<Customer> on the CustomerGateway

Sav4ykVladzislau committed:
Change input on the post method

Sav4ykVladzislau committed:
Create new endpoint /customer and create test

Sav4ykVladzislau committed:
Delete endpoint /customer and test

Sav4ykVladzislau committed:
Add error message in customer-form.html

Sav4ykVladzislau committed:
Merge branch 'main' into TGBOT-289-customer-methods

Sav4ykVladzislau committed:
Moved "spring-boot-starter-web" dependency in the child pom

Sav4ykVladzislau committed:
Create HTTPCustomerGateway

Sav4ykVladzislau committed:
Change customer-form.html add action

Sav4ykVladzislau committed:
Add dependency on the separate module

Sav4ykVladzislau committed:
Delete wrong class

Sav4ykVladzislau committed:
Create HttpCustomerGateway and change CustomerController

Sav4ykVladzislau committed:
Add updateCustomer method from interface CustomerGateway

Sav4ykVladzislau committed:
Create tests root

Sav4ykVladzislau committed:
Add new implements method and code refactoring

Sav4ykVladzislau committed:
Code refactoring

Sav4ykVladzislau committed:
Add button "update"

Sav4ykVladzislau committed:
Change method "save"->"update"

Sav4ykVladzislau committed:
Add method updateCustomer for interface CustomerGateway

Sav4ykVladzislau committed:
Change method signature

Sav4ykVladzislau committed:
Change methods

Sav4ykVladzislau committed:
Change methods

Sav4ykVladzislau committed:
Add action endpoint on the form

Sav4ykVladzislau committed:
Add method update on the gateway interface

Sav4ykVladzislau committed:
Add component scan annotations

Sav4ykVladzislau committed:
Add another URL address and extracting the code of the creator HttpClient

Sav4ykVladzislau committed:
Returning the controller to its initial state

Sav4ykVladzislau committed:
Create EndpointConfig

Sav4ykVladzislau committed:
Changed redirect address

Sav4ykVladzislau committed:
Changed the method of calling CustomerGateway

Sav4ykVladzislau committed:
Changed the method of calling EndpointConfig
and added encapsulation of the endpoint address

Sav4ykVladzislau committed:
added unit tests and test context configuration class

Sav4ykVladzislau committed:
Changed mock http address

Sav4ykVladzislau committed:
Changed declare EndpointConfig

Sav4ykVladzislau committed:
Merge remote-tracking branch 'origin/main' into TGBOT-296-clone-create-http-client-customer-gateway-implementation

Sav4ykVladzislau committed:
Merge branch 'main' into TGBOT-296-clone-create-http-client-customer-gateway-implementation

Sav4ykVladzislau committed:
Changed context import and fix build

Sav4ykVladzislau committed:
Changed class declaration CustomerGateway

Sav4ykVladzislau committed:
Merge remote-tracking branch 'origin/main' into TGBOT-296-clone-create-http-client-customer-gateway-implementation

Sav4ykVladzislau committed:
Delete mock class

Sav4ykVladzislau committed:
Change annotations on the class EndpointConfig

Sav4ykVladzislau committed:
Change redirect page in the test result

max-java commented on http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java:
```diff
@@ -0,0 +1,93 @@
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(endpointConfig.getRestApiUrl()))
+                .POST(HttpRequest.BodyPublishers
+                        .ofString(String.format(
+                                "{\"customerId\":\"%s\"\"name\":\"%s\",\"city\":\"%s\",\"phoneNumber\":\"%s\",\"email\":\"%s\"}",
+                                customer.customerId(), customer.name(), customer.city(), customer.phoneNumber(), customer.email())))
```
this should be posted as body, maybe map from string ? 
good example here https://stackoverflow.com/questions/66388231/json-object-to-body-publisher

Sav4ykVladzislau replied:
I did

max-java commented on web-ui/pom.xml:
```diff
@@ -21,6 +21,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.tutrit</groupId>
+            <artifactId>http-client</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
```
this doesn't  belong here

Sav4ykVladzislau replied:
I did

max-java commented on web-ui/src/main/java/com/tutrit/webclient/WebClientApplication.java:
```diff
@@ -2,8 +2,10 @@
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@ComponentScan(basePackages = {"com.tutrit"})
```
redundant, could be removed

Sav4ykVladzislau replied:
I did

max-java commented on web-ui/src/main/java/com/tutrit/webclient/controller/CustomerController.java:
```diff
@@ -14,20 +14,17 @@
 @Controller
 public class CustomerController {
-    @Autowired(required = false)
+
```
changes to web-ui module are out of scope of current issue

Sav4ykVladzislau replied:
I agree with you. I reverted this changes

max-java commented on web-ui/src/main/resources/templates/customer-form.html:
```diff
@@ -1,14 +1,14 @@
 <!DOCTYPE html>
 <html lang="en" xmlns:th="http://www.thymeleaf.org">
```
changes to web-ui module are out of scope of current issue

Sav4ykVladzislau replied:
yes, i agree

max-java reviewed (changes requested):
ptal,

Sav4ykVladzislau committed:
Change gateway methods and test

Sav4ykVladzislau committed:
Remove wrong dependency in web-ui pom

Sav4ykVladzislau committed:
Returned http-client pom file to the initial state

Sav4ykVladzislau committed:
remove redundant mapper

Refactoring methods on the HttpCustomerGateway and move tests into working

remove bean ObjectMapper from HttpClientConfig

add finalise field and add object mapper bin

Add dependency http-client -> web-ui

finalized value into HttpCustomerGateway

Change Value annotation

Change the default call address

Change method to map body

[TGBOT-323] Make an index html and home page controller (#19)

* Create HomePageController

* Add empty line

Sav4ykVladzislau committed:
remove redundant mapper

Sav4ykVladzislau committed:
Refactoring methods on the HttpCustomerGateway and move tests into working

Sav4ykVladzislau committed:
remove bean ObjectMapper from HttpClientConfig

Sav4ykVladzislau committed:
add finalise field and add object mapper bin

Sav4ykVladzislau committed:
Add dependency http-client -> web-ui

Sav4ykVladzislau committed:
finalized value into HttpCustomerGateway

Sav4ykVladzislau committed:
Change Value annotation

Sav4ykVladzislau committed:
Change the default call address

Sav4ykVladzislau committed:
Change method to map body

Sav4ykVladzislau committed:
Merge remote-tracking branch 'origin/TGBOT-296-clone-create-http-client-customer-gateway-implementation' into TGBOT-296-clone-create-http-client-customer-gateway-implementation

max-java commented on http-client/pom.xml:
```diff
@@ -26,7 +26,17 @@
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.tutrit</groupId>
+            <artifactId>web-ui</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>compile</scope>
```
I doubt we need it here - that could lead to circular dependencies

max-java commented on web-ui/src/main/java/com/tutrit/webclient/controller/HomePageController.java:
```diff
@@ -0,0 +1,18 @@
+package com.tutrit.webclient.controller;
```
doesn't seem to be related to #16, does it?

max-java commented on web-ui/src/main/resources/templates/starter.html:
```diff
@@ -0,0 +1,354 @@
+<!DOCTYPE html>
```
doesn't seem to be related to https://github.com/Tutritp01/TgBot_WebClient/pull/16, does it?

max-java reviewed (commented):
minor comments.

Sav4ykVladzislau committed:
Change default address

Sav4ykVladzislau committed:
Change default address

Sav4ykVladzislau committed:
Delete HomePageController.java

removed file which is not relevant to this task

Sav4ykVladzislau committed:
Delete customer-form.html

remove form

Sav4ykVladzislau committed:
[TGBOT-296] Resolve conflict and merge current branch on main

max-java reviewed (approved):
lgfm
